neuro_san/session/async_direct_agent_session.py

```diff
@@ -173,18 +174,21 @@ async def streaming_chat(self, request_dict: Dict[str, Any]) -> Generator[Dict[s
         # The generator below will asynchronously block waiting for
         # chat.ChatMessage dictionaries to come back asynchronously from the submit()
         # above until there are no more from the input.
-        generator = self.invocation_context.get_queue()
-        async for message in generator:
-
-            response_dict: Dict[str, Any] = copy(template_response_dict)
-            if message_filter.allow(message):
-                # We expect the message to be a dictionary form of chat.ChatMessage
-                if message_processor is not None:
-                    message_type: ChatMessageType = message.get("type")
-                    # Can modify message
-                    await message_processor.async_process_message(message, message_type)
-                response_dict["response"] = message
-                yield response_dict
+        queue_generator = self.invocation_context.get_queue()
+        try:
+            async for message in queue_generator:
+                response_dict: Dict[str, Any] = copy(template_response_dict)
+                if message_filter.allow(message):
+                    # We expect the message to be a dictionary form of chat.ChatMessage
+                    if message_processor is not None:
+                        message_type: ChatMessageType = message.get("type")
+                        # Can modify message
+                        await message_processor.async_process_message(message, message_type)
+                    response_dict["response"] = message
+                    yield response_dict
+        finally:
+            with contextlib.suppress(Exception):
+                await chat_session.delete_resources()
 
```

Comment:
Change of logic:
1. We tell underlying chat_session to delete its resources since we are done with this request;
2. We do this in "finally" block so this releasing of resources happens in any case, including getting our async generator (which is implicitly constructed by lines 179-188 and returned by this method) interrupted by caller-side "aclose" method.
3. And we suppress all exceptions while deleting resources to keep things quieter.